Can't open a directory called specs

i'm not sure what's going on here. it only happens on this directory, but if i make another directory called specs, atom opens.

i open atom by going to the folder in the terminal and typing `atom` or `atom .`.

![screen shot 2015-01-15 at 11 13 51 am](https://cloud.githubusercontent.com/assets/643505/5764594/2e7d96a6-9ca8-11e4-86db-63b5308f6a83.png)

what else do you guys need?

```
index.js:80 Error: /Users/jong/Workspace/sunsetmtn84/specs is a directory
  at new File (/Applications/Atom.app/Contents/Resources/app/node_modules/pathwatcher/lib/file.js:49:15)
  at TextBuffer.module.exports.TextBuffer.setPath (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:279:21)
  at new TextBuffer (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:100:14)
  at Function.module.exports.Serializable.deserialize (/Applications/Atom.app/Contents/Resources/app/node_modules/serializable/lib/serializable.js:72:22)
  at DeserializerManager.module.exports.DeserializerManager.deserialize (/Applications/Atom.app/Contents/Resources/app/src/deserializer-manager.js:56:29)
  at /Applications/Atom.app/Contents/Resources/app/src/project.js:128:35
  at Array.map (native)
  at Project.module.exports.Project.deserializeParams (/Applications/Atom.app/Contents/Resources/app/src/project.js:127:39)
  at Function.module.exports.Serializable.deserialize (/Applications/Atom.app/Contents/Resources/app/node_modules/serializable/lib/serializable.js:58:25)
  at DeserializerManager.module.exports.DeserializerManager.deserialize (/Applications/Atom.app/Contents/Resources/app/src/deserializer-manager.js:56:29)
  at Atom.module.exports.Atom.deserializeProject (/Applications/Atom.app/Contents/Resources/app/src/atom.js:725:52)
  at Atom.module.exports.Atom.deserializeEditorWindow (/Applications/Atom.app/Contents/Resources/app/src/atom.js:753:12)
  at Atom.module.exports.Atom.startEditorWindow (/Applications/Atom.app/Contents/Resources/app/src/atom.js:603:12)
  at Object.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/window-bootstrap.js:12:8)
  at Object.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/window-bootstrap.js:23:4)
  at Module._compile (module.js:468:26)
  at Object.Module._extensions..js (module.js:486:10)
  at Module.load (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:318:12)
  at Module.require (module.js:373:17)
  at require (module.js:392:17)
  at window.onload (file:///Applications/Atom.app/Contents/Resources/app/static/index.js:66:25)
index.js:80 window.onload
```

--- a/package.json
+++ b/package.json
@@ -46,7 +46,7 @@
     "nslog": "^1.0.1",
     "oniguruma": "^3.0.6",
     "optimist": "0.4.0",
-    "pathwatcher": "^2.6.0",
+    "pathwatcher": "^2.6.1",
     "property-accessors": "^1",
     "q": "^1.0.1",
     "random-words": "0.0.1",

--- a/spec/project-spec.coffee
+++ b/spec/project-spec.coffee
@@ -50,6 +50,19 @@ describe "Project", ->
         deserializedProject = atom.project.testSerialization()
         expect(deserializedProject.getBuffers().length).toBe 0
 
+    it "does not deserialize buffers when their path is inaccessible", ->
+      pathToOpen = path.join(temp.mkdirSync(), 'file.txt')
+      fs.writeFileSync(pathToOpen, '')
+
+      waitsForPromise ->
+        atom.project.open(pathToOpen)
+
+      runs ->
+        expect(atom.project.getBuffers().length).toBe 1
+        fs.chmodSync(pathToOpen, '000')
+        deserializedProject = atom.project.testSerialization()
+        expect(deserializedProject.getBuffers().length).toBe 0
+
   describe "when an editor is saved and the project has no path", ->
     it "sets the project's path to the saved file's parent directory", ->
       tempFile = temp.openSync().path

--- a/spec/project-spec.coffee
+++ b/spec/project-spec.coffee
@@ -38,7 +38,7 @@ describe "Project", ->
         expect(deserializedProject.getBuffers().length).toBe 0
 
 
-    it "does not deserialize buffers when the path is a directory", ->
+    it "does not deserialize buffers when their path is a directory that exists", ->
       pathToOpen = path.join(temp.mkdirSync(), 'file.txt')
 
       waitsForPromise ->

--- a/spec/project-spec.coffee
+++ b/spec/project-spec.coffee
@@ -37,6 +37,19 @@ describe "Project", ->
         deserializedProject.getBuffers()[0].destroy()
         expect(deserializedProject.getBuffers().length).toBe 0
 
+
+    it "does not deserialize buffers when the path is a directory", ->
+      pathToOpen = path.join(temp.mkdirSync(), 'file.txt')
+
+      waitsForPromise ->
+        atom.project.open(pathToOpen)
+
+      runs ->
+        expect(atom.project.getBuffers().length).toBe 1
+        fs.mkdirSync(pathToOpen)
+        deserializedProject = atom.project.testSerialization()
+        expect(deserializedProject.getBuffers().length).toBe 0
+
   describe "when an editor is saved and the project has no path", ->
     it "sets the project's path to the saved file's parent directory", ->
       tempFile = temp.openSync().path

--- a/src/project.coffee
+++ b/src/project.coffee
@@ -226,8 +226,7 @@ class Project extends Model
 
     if filePath?
       try
-        fileDescriptor = fs.openSync(filePath, 'r+')
-        fs.closeSync(fileDescriptor)
+        fs.closeSync(fs.openSync(filePath, 'r+'))
       catch error
         # allow ENOENT errors to create an editor for paths that dont exist
         throw error unless error.code is 'ENOENT'

--- a/src/project.coffee
+++ b/src/project.coffee
@@ -67,11 +67,14 @@ class Project extends Model
 
   deserializeParams: (params) ->
     params.buffers = _.compact params.buffers.map (bufferState) ->
-      try
-        atom.deserializers.deserialize(bufferState)
-      catch error
-        # Ignore buffers whose previous paths are now folders
-        throw error unless error.code is 'EISDIR'
+      # Check that buffer's file path is accessible
+      if bufferState.filePath
+        try
+          fs.closeSync(fs.openSync(bufferState.filePath, 'r+'))
+        catch error
+          return unless error.code is 'ENOENT'
+
+      atom.deserializers.deserialize(bufferState)
     params
 
   ###

--- a/src/project.coffee
+++ b/src/project.coffee
@@ -66,7 +66,12 @@ class Project extends Model
     buffers: _.compact(@buffers.map (buffer) -> buffer.serialize() if buffer.isRetained())
 
   deserializeParams: (params) ->
-    params.buffers = params.buffers.map (bufferState) -> atom.deserializers.deserialize(bufferState)
+    params.buffers = _.compact params.buffers.map (bufferState) ->
+      try
+        atom.deserializers.deserialize(bufferState)
+      catch error
+        # Ignore buffers for files that are now folders
+        throw error unless error.code is 'EISDIR'
     params
 
 

--- a/src/project.coffee
+++ b/src/project.coffee
@@ -70,7 +70,7 @@ class Project extends Model
       try
         atom.deserializers.deserialize(bufferState)
       catch error
-        # Ignore buffers for files that are now folders
+        # Ignore buffers whose previous paths are now folders
         throw error unless error.code is 'EISDIR'
     params
 

--- a/src/project.coffee
+++ b/src/project.coffee
@@ -74,7 +74,6 @@ class Project extends Model
         throw error unless error.code is 'EISDIR'
     params
 
-
   ###
   Section: Event Subscription
   ###

